code/obj/item/beartrap.dm

```diff
@@ -84,18 +74,21 @@
 			src.anchored = FALSE
 		return
 
-	proc/triggered(mob/target as mob)
+	proc/triggered(mob/target)
 		if (!src || !src.armed)
 			return
 
-		if (target && ishuman(target))
-			var/mob/living/carbon/human/H = target
-			logTheThing(LOG_COMBAT, H, "stood on a [src] at [log_loc(src)].")
-			H.changeStatus("stunned", 4 SECONDS)
-			H.force_laydown_standup()
-			random_brute_damage(H, 50, 0)
-			take_bleeding_damage(H, null, 15, DAMAGE_CUT)
-			H.UpdateDamageIcon()
+		if (target && isliving(target) && !isintangible(target))
+			var/mob/living/M = target
+			logTheThing(LOG_COMBAT, M, "stood on a [src] at [log_loc(src)].")
+			if(istype(M, /mob/living/critter/bear))
+				M.death()
+			else
+				M.changeStatus("stunned", 4 SECONDS)
+				M.force_laydown_standup()
+				random_brute_damage(M, 50, 0)
+				take_bleeding_damage(M, null, 15, DAMAGE_CUT)
```

Comment:
Maybe need at least another check here, there's a small issue where critters like ghost critters, as well as random NPC ones like cockroaches, can trigger the beartrap which could be annoying when you're just wanting it to trigger on people. Would be more applicable on nukie rounds, not sure though

Reply:
hmm ghost critter check sounds good, I think it should be fine activating on normal mobs though. Its pretty sensible.